### PR TITLE
Remove plan vm from PlanBoxView and PlanCardView

### PR DESCRIPTION
**Pull request by keanecjy**

### CODE REVIEW

File: TourMate/TourMate/Frontend/Plan/Views/PlansViews/Components/PlanCardView.swift [hunk 1]
```diff
@@ -7,29 +7,38 @@
...
+    init(planUpvoteViewModel: PlanUpvoteViewModel, planName: String,
+         planStatus: PlanStatus, startDateTime: DateTime, endDateTime: DateTime,
+         date: Date) {
+        self.planUpvoteViewModel = planUpvoteViewModel
+        self.planName = planName
+        self.planStatus = planStatus
+        self.startDateTime = startDateTime
+        self.endDateTime = endDateTime
         self.date = date
     }
```
Comment (tanruiquan) [inline TourMate/TourMate/Frontend/Plan/Views/PlansViews/Components/PlanCardView.swift]: Would it be better if it is just this?
```suggestion
    init(planUpvoteViewModel: PlanUpvoteViewModel, plan: Plan) {
        self.planUpvoteViewModel = planUpvoteViewModel
        self.plan = plan
    }
```
---
Then if we want to access any of the `Plan` properties we can for example, 
```
Text(plan.name).font(.headline)
```

**Review by tanruiquan — APPROVED** [review]
LGTM, thanks for doing the `DateUtil` too 💯

File: TourMate/TourMate/Frontend/Plan/Views/PlansViews/Components/PlanCardView.swift (hunk 1 above)
Comment (keanecjy) [reply TourMate/TourMate/Frontend/Plan/Views/PlansViews/Components/PlanCardView.swift]: yeah i think its better as well, but prof seems to dislike passing the data object in

Comment (keanecjy) [reply TourMate/TourMate/Frontend/Plan/Views/PlansViews/Components/PlanCardView.swift]: anyway i will change it

**Pull request merged**